File: CourierAmexAPI/CourierAmex/Models/UserModel.cs
namespace CourierAmex.Models
{
    public class UserModel : BaseEntityModel<string>
    {
        public required string Name { get; set; }
        public required string Lastname { get; set; }
        public required string Username { get; set; }
        public string? Email { get; set; }
        public string? Mobile { get; set; }
        public string? Phone { get; set; }
        public string? Office { get; set; }
        public int? CountryId { get; set; }
        public short? StateId { get; set; }
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? City { get; set; }
        public string? Zip { get; set; }
        public byte? Gender { get; set; }
        public int CompanyId { get; set; }
        public byte OperationType { get; set; }
        public byte SessionTimeout { get; set; }
        public long? DateOfBirth { get; set; }
        public long? LastLoginDate { get; set; }
        public bool ChangePassword { get; set; }
        public CompanyModel? Company { get; set; }
        public IEnumerable<RoleModel>? Roles { get; set; }
        public IEnumerable<PermissionModel>? Permissions { get; set; }
    }

    public class ChangePasswordModel
    {
        public required string Id { get; set; }
        public required string CurrentPassword { get; set; }
        public required string NewPassword { get; set; }
    }
}

File: CourierAmexClient/src/app/features/user/models/user.model.ts
import { BaseEntity } from "@app/models";
import { RoleModel } from "./role.model";
import { DEFAULT_STATUS_ACTIVE, DEFAULT_USER_SESSION_TIMEOUT } from "@app/@shared";

export interface UserModel extends BaseEntity<string> {
  name: string;
  lastname: string;
  username: string;
  email: string;
  mobile: string;
  phone: string;
  office: string;
  countryId?: number;
  stateId?: number;
  addressLine1: string;
  addressLine2: string;
  dateOfBirth?: number;
  dateOfBirthValue?: string;
  city: string;
  zip: string;
  gender?: number;
  operationType: number;
  companyId?: number;
  sessionTimeout?: number;
  tempPassword?: string;
  
  roles: RoleModel[];
}

export const newUserModel = {
  id: '',
  name: '',
  lastname: '',
  username: '',
  email: '',
  mobile: '',
  phone: '',
  office: '',
  countryId: 0,
  stateId: 0,
  addressLine1: '',
  addressLine2: '',
  city: '',
  zip: '',
  gender: 0,
  companyId: undefined,
  dateOfBirth: undefined,
  status: DEFAULT_STATUS_ACTIVE,
  operationType: 0,
  sessionTimeout: DEFAULT_USER_SESSION_TIMEOUT,
  
  roles: []
}
